### PR TITLE
Upgrade Aesara and Aeppl dependencies

### LINKED ISSUE #5681: Aesara warning on using dict in updates
```
/Users/twiecki/miniforge3/envs/pymc4b5/lib/python3.10/site-packages/pymc/aesaraf.py:996: UserWarning: The parameter 'updates' of aesara.function() expects an OrderedDict, got <class 'dict'>. Using a standard dictionary here results in non-deterministic behavior. You should use an OrderedDict if you are using Python 2.7 (collections.OrderedDict for older python), or use a list of (shared, update) pairs. Do not just convert your dictionary to this type before the call as the conversion will still be non-deterministic.
```

Code:
```
import numpy as np
import arviz as a
import matplotlib.pyplot as plt
import seaborn as sns
import pandas as pd

# PyMC Imports
import pymc as pm4 # PyMC 4.0

# Aesara and Theano imports
import aesara.tensor as at # used by PyMC 4.0
import aesara

data = pd.read_csv(pm4.get_data("radon.csv"))
county_names = data.county.unique()

data["log_radon"] = data["log_radon"].astype(aasara.config.floatX)

county_idx, counties = pd.factorize(data.county)
coords = {
    "county": counties,
    "obs_id": np.arange(len(county_idx)),
}

def build_model(pm, **kwargs):
    with pm.Model(coords=coords) as hierarchical_model:
        mu_a = pm.Normal("mu_a", mu=0.0, sigma=10)
        sigma_a = pm.HalfNormal("sigma_a", 5.0)
        mu_b = pm.Normal("mu_b", mu=0.0, sigma=10)
        sigma_b = pm.HalfNormal("sigma_b", 5.0)
        a = pm.Normal("a", dims="county") * sigma_a + mu_a
        b = pm.Normal("b", dims="county") * sigma_b + mu_b
        eps = pm.HalfNormal("eps", 5.0)
        radon_est = a[county_idx] + b[county_idx] * data.floor.values
        radon_like = pm.Normal(
            "radon_like", mu=radon_est, sigma=eps, observed=data.log_radon, 
            dims="obs_id"
        )
        
    return hierarchical_model

model_pymc4 = build_model(pm4)
with model_pymc4:
    idata_pymc4 = pm4.sample()

### PR DESCRIPTION
**Pull request by michaelosthege**
Let's see if this works.

There's a decent chance that mypy checks are going to fail.

Closes #5681

### CODE REVIEW

**Comment by michaelosthege** [comment]
Anybody feel free to take over. I don't have time to work on this.

**Comment by ricardoV94** [comment]
Probably related to sparse matrices in the CAR distribution, as those were changed quite a lot in Aesara

**Comment by ricardoV94** [comment]
Failing tests are due to https://github.com/aesara-devs/aesara/issues/913

**Comment by ricardoV94** [comment]
Tests are passing at last

File: pymc/distributions/mixture.py [hunk 1]
```diff
@@ -421,12 +412,6 @@ def marginal_mixture_logcdf(op, value, rng, weights, *components, **kwargs):
 
     mix_logcdf = at.logsumexp(at.log(weights) + components_logcdf, axis=-1)
 
-    # Squeeze stack dimension
```
Comment (twiecki) [inline pymc/distributions/mixture.py:424]: Nice to be able to remove all this.

**Review by twiecki — APPROVED** [review]

**Pull request merged**